In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
df=pd.read_csv('Cleaned_Data.csv')

In [3]:
df

Name  Original_price  \
0                      OYO 22085 Raj Banjara White House          2459.0   
1                        Collection O 42514 Venkat Nagar          3158.0   
2                 Capital O 44354 Jubilee Banjara Suites          1778.0   
3                            OYO 35487 Hotel Meena Grand          2090.0   
4                              OYO 16018 Rainbow Saphire          2607.0   
5                    Collection O 46198 New Sree Krishna          3964.0   
6                            OYO 14224 Happy Guest house          2373.0   
7                          OYO 62547 Dream Connect Stays          1941.0   
8                                  OYO 11376 Golden Park          2205.0   
9                                  OYO 11316 Athithi Inn          2178.0   
10                                  OYO 10272 Hotel Ruma          1777.0   
11                                    OYO 18725 Sr Grand          1910.0   
12                               OYO 18302 Pride Banjara          2260.0   
13                         OYO 983 Hotel Surya Residency          2811.0   
14             Capital O 46832 Gaayatri Residency Deluxe          2098.0   
15                        OYO 4025 Hotel Meredian Orchid          1865.0   
16                        OYO Townhouse 045 Lakdi ka Pul          5129.0   
17                             OYO 41751 Hotel Sabharwal          1416.0   
18                               OYO 47332 Hotel Dolphin          2509.0   
19                                   OYO 30744 The  Gate          2140.0   
20                             OYO 37906 New Asian Lodge          2420.0   
21                                OYO 26562 Dazzle Stays          1730.0   
22            SilverKey Executive Stays 27773 IAS Colony          1639.0   
23                        OYO 1410 Country Club Begumpet          4417.0   
24                        OYO 22907 Hotel Holiday Deluxe          2407.0   
25                                     OYO 47727 S Grand          1427.0   
26                             Capital O 6422 Luxor Park          2737.0   
27                            OYO 9302 Hotel Baseraa Inn          2081.0   
28                          OYO 8941 Hotel Golden Leaves          2449.0   
29                    OYO Flagship 14348 Apollo Hospital          2746.0   
...                                                  ...             ...   
11952                               OYO 46744 Hotel Yash          1481.0   
11953                             OYO 62553 Suvee Palace          1294.0   
11954                              OYO 61639 Red Carmine          1777.0   
11955            OYO 44324 Comfort Stay Near Saket Metro          2989.0   
11956              OYO Home Luxe 30142 Fabulous Farmstay          2548.0   
11957                 OYO 22321 Nitin Palace Guest House          2822.0   
11958                       OYO Townhouse 002 Green Park          7806.0   
11959      Capital O 7075 The Oakland Plaza Hotel Deluxe          3078.0   
11960                           OYO 14789 Kempinski Wide          3534.0   
11961                               OYO 63837 Ambica Inn          2729.0   
11962                OYO 37512 Spacious Stay Saket Metro          1981.0   
11963                      OYO 60334 Skp Bed & Breakfast          2225.0   
11964                            OYO 64271 Singh Hotel 2          1140.0   
11965                      OYO 37055 M R Residency Saver          1481.0   
11966                Capital O 712 Defence Colony Deluxe          3738.0   
11967          OYO Home 47965 Cozy Stay Ignou Road Saket          2720.0   
11968  OYO Home 47964 Comfortable  Stay Ignou Road Saket          2720.0   
11969                              OYO 44400 Retreat Inn          1878.0   
11970      OYO Flagship 36485 Smg Heritage Nehru Enclave          1777.0   
11971                     Capital O 39462 Westyard Hotel          2863.0   
11972                                   OYO 40657 Aastha          1372.0   
11973                   

In [4]:
df.head()

Name  Original_price  Offer  Final_price  \
0       OYO 22085 Raj Banjara White House          2459.0   32.0       1677.0   
1         Collection O 42514 Venkat Nagar          3158.0   32.0       2153.0   
2  Capital O 44354 Jubilee Banjara Suites          1778.0   32.0       1213.0   
3             OYO 35487 Hotel Meena Grand          2090.0   32.0       1425.0   
4               OYO 16018 Rainbow Saphire          2607.0   32.0       1777.0   

   Distance  Rating  Number_of_Rating    Summary       Location       City  
0       1.7     4.4             201.0  Very Good  Banjara Hills  Hyderabad  
1       1.6     4.4             879.0  Very Good  Banjara Hills  Hyderabad  
2       1.1     4.4            1258.0  Very Good  Banjara Hills  Hyderabad  
3       4.0     4.0             575.0  Very Good  Banjara Hills  Hyderabad  
4       4.6     4.7             978.0  Excellent  Banjara Hills  Hyderabad

In [5]:
df.tail()

Name  Original_price  Offer  Final_price  \
11977             OYO 472 Hotel Sareen          2389.0   35.0       1552.0   
11978  OYO 43002 The Royal Town Deluxe          4522.0   32.0       3083.0   
11979       OYO 48664 Gour White House          2526.0   32.0       1723.0   
11980    OYO 12532 Imperial Inn Deluxe          7544.0   32.0       5144.0   
11981        OYO 13235 The Golden Door          6855.0   32.0       4674.0   

       Distance  Rating  Number_of_Rating    Summary     Location   City  
11977       2.0     3.7              90.0       Good  Nehru Place  Delhi  
11978       1.0     3.6             114.0       Good  Nehru Place  Delhi  
11979       2.8     4.3             245.0  Very Good  Nehru Place  Delhi  
11980       1.3     4.8             271.0  Excellent  Nehru Place  Delhi  
11981       2.8     4.5             173.0  Excellent  Nehru Place  Delhi

In [6]:
df.shape

(11982, 10)

- The total data includes 11982 hotels.

In [7]:
df.keys()

Index(['Name', 'Original_price', 'Offer', 'Final_price', 'Distance', 'Rating',
       'Number_of_Rating', 'Summary', 'Location', 'City'],
      dtype='object')

In [8]:
df.dtypes

Name                 object
Original_price      float64
Offer               float64
Final_price         float64
Distance            float64
Rating              float64
Number_of_Rating    float64
Summary              object
Location             object
City                 object
dtype: object

# Understand Categorical Data

- Columns containing categorical columns:
    - City
    - Location
    - Summary

## City

In [9]:
df['City'].unique()

array(['Hyderabad', 'Bangalore', 'Chennai', 'Mumbai', 'Delhi'],
      dtype=object)

In [10]:
df['City'].nunique()

5

In [11]:
df['City'].value_counts()

Delhi        3602
Bangalore    2720
Hyderabad    2440
Chennai      1760
Mumbai       1460
Name: City, dtype: int64

- We can see from the above data that, Delhi has the maximum number of hotels compared to the rest.
- While Chennai has the least number of hotels.

## Location

In [13]:
df['Location'].unique()

array(['Banjara Hills', 'Jubilee Hills', 'Kondapur', 'Hitech City',
       'Begumpet', 'Mehdipatnam', 'Kachiguda Railway Station',
       'Secunderabad Railway Station', 'LB Nagar', 'Kukatpally', 'Uppal',
       'Gachibowli', 'Ameerpet', 'Rajiv Gandhi International Airport',
       'Lakdikapool', 'Shamshabad', 'Himayathnagar', 'Madhapur',
       'Miyapur', 'Nampally', 'Indiranagar', 'Domlur', 'Majestic',
       'Jayanagar', 'Whitefield', 'Majestic City Railway Station',
       'Kempagowda International Airport', 'Kalyan Nagar', 'Rajaji Nagar',
       'Hebbal', 'HSR Layout', 'Yelahanka', 'MG Road', 'KR Puram',
       'Jp Nagar', 'Electronic City', 'Yeshwanthpur', 'Koramangala',
       'BTM 2ndStage', 'Marathahalli', 'Sholinganallur', 'Velachery',
       'Egmore', 'Chennai Central Railway Station', 'Guindy',
       'Vadapalani', 'Mogappair', 'Porur', 'Thiruvanmiyu',
       'Chennai International Airport', 'East Coast Road', 'Tnagar',
       'Nungambakkam', 'Koyambedu', 'Chennai Egmore Ra

In [14]:
df['Location'].nunique()

100

In [15]:
df['Location'].value_counts()

Mahipalpur                            320
Laxminagar                            260
New Delhi Railway Station             260
Sector 7 Dwaraka                      240
MG Road                               240
Saket                                 220
Hazrat Nizammudin Railway Station     200
Jayanagar                             200
Paschim Vihar                         200
Karol Bagh                            200
Paharganj                             200
Domlur                                200
Karol Bagh Metro Station              200
Jubilee Hills                         180
Majestic                              180
Indira Gandhi Airport                 180
Indiranagar                           180
BTM 2ndStage                          180
Kashmiri Gate ISBT                    180
Pitampura                             180
Jp Nagar                              180
Banjara Hills                         180
Majestic City Railway Station         180
Koramangala                       

- There are 100 locations.
- Mahipalpur has the most number of hotels.

## Summary

In [20]:
df['Summary'].unique()

array(['Very Good', 'Excellent', 'Good', 'Fair', 'Yet to be summarised',
       'Fabulous'], dtype=object)

In [21]:
df['Summary'].nunique()

6

In [22]:
df['Summary'].value_counts()

Very Good               4982
Good                    3067
Excellent               1743
Yet to be summarised    1122
Fair                     980
Fabulous                  88
Name: Summary, dtype: int64

- It can be seen that the summary has been divided into 6.
- 4982 hotels from the data fall under 'Very Good'.
- While only 88 hotels are 'Fabulous'.

# Understand Numerical Data

## Original_price

In [24]:
df['Original_price'].describe()

count    11982.000000
mean      2895.721249
std       1870.412218
min        602.000000
25%       1797.000000
50%       2299.000000
75%       3305.000000
max      32452.000000
Name: Original_price, dtype: float64

- Minimum Original_price for the data is Rs.602
- Maximum Original_price for the data is Rs.32452
- Mean value of Original_price is Rs.2895.7212
- Median value of Original_price is Rs.2299


## Offer

In [26]:
df['Offer'].describe()

count    11982.000000
mean        34.331247
std          5.889520
min         32.000000
25%         32.000000
50%         32.000000
75%         32.000000
max         71.000000
Name: Offer, dtype: float64

- Minimum Offer for the data is 32%
- Maximum Offer for the data is 71%
- Mean value of Offer is 34.33%
- Median value of Offer is 32%

## Final_price

In [28]:
df['Final_price'].describe()

count    11982.000000
mean      1908.409698
std       1272.557859
min        411.000000
25%       1212.000000
50%       1506.000000
75%       2135.250000
max      20651.000000
Name: Final_price, dtype: float64

- Minimum Final_price for the data is Rs.411
- Maximum Final_price for the data is Rs.20651
- Mean value of Final_price is Rs.1908.409698
- Median value of Final_price is Rs.1506

## Distance

In [30]:
df['Distance'].describe()

count    11982.000000
mean         6.155658
std          3.386736
min          0.000000
25%          3.600000
50%          6.200000
75%          8.300000
max         20.000000
Name: Distance, dtype: float64

- Minimum Distance for the data is 0.0 km
- Maximum Distance for the data is 20 km
- Mean value of Distance is 6.155658 km
- Median value of Distance is 6.2 km

## Rating

In [32]:
df['Rating'].describe()

count    11982.000000
mean         3.670547
std          1.243398
min          0.000000
25%          3.600000
50%          4.000000
75%          4.300000
max          5.000000
Name: Rating, dtype: float64

- Minimum Rating for the data is 0
- Maximum Rating for the data is 5
- Mean value of Rating is 3.67
- Median value of Rating is 4

## Number of Rating

In [34]:
df['Number_of_Rating'].describe()

count    11982.000000
mean       588.801536
std        807.178290
min          0.000000
25%        116.000000
50%        290.000000
75%        719.000000
max       6943.000000
Name: Number_of_Rating, dtype: float64

- Minimum Number of Rating for the data is 0 Ratings
- Maximum Number of Rating for the data is 6943 Ratings
- Mean value of Number of Rating is 588.801536 Ratings
- Median value of Number of Rating is 290 Ratings